apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx

```diff
@@ -10,22 +10,33 @@ import {useSelectedWallet} from '../../../../SelectedWallet'
 import {COLORS} from '../../../../theme'
 import {useTokenInfo} from '../../../../yoroi-wallets/hooks'
 import {Quantities} from '../../../../yoroi-wallets/utils'
+import {getPriceImpactStatus, usePriceImpactStatusTheme} from '../../common/helpers'
 import {LiquidityPool} from '../../common/LiquidityPool/LiquidityPool'
 import {PoolIcon} from '../../common/PoolIcon/PoolIcon'
 import {useStrings} from '../../common/strings'
+import {useSwapForm} from '../../common/SwapFormProvider'
 import {SwapInfoLink} from '../../common/SwapInfoLink/SwapInfoLink'
 
 export const TransactionSummary = () => {
   const strings = useStrings()
   const wallet = useSelectedWallet()
   const {orderData} = useSwap()
-  const {amounts, selectedPoolCalculation: calculation} = orderData
+  const {
+    limitPrice: {displayValue: limitDisplayValue},
+  } = useSwapForm()
+  const {amounts, selectedPoolCalculation: calculation, calculations} = orderData
   const {openModal} = useModal()
 
   // should never happen
   if (!calculation) throw new Error('No selected pool calculation')
   const {pool, cost} = calculation
 
+  const priceImpact = calculations[0]?.prices.priceImpact
```

Comment:
calculation (has all the calculations), it should use on summary, calculation, which is the selected calculation.